test/ts/prect-test.tsx

```diff
@@ -1,32 +1,425 @@
-import { h, render, Component } from 'preact';
+import { h, render, Component, SFC, ChildContextProvider, ComponentProps, ComponentClass, VNode } from 'preact';
 
-interface DummyProps {
-	initialInput: string;
-}
+// import 'preact/devtools';
 
-interface DummyState {
-	input: string;
-}
+// #Components and StatellesComponents
+type DummyProps = { initialInput: string; }
+type DummyState = { input: string; }
 
 class DummyComponent extends Component<DummyProps, DummyState> {
+	// static members cannot be typechecked by inferenece
+	static defaultProps: DummyProps = {
+		initialInput: 'hmm'
+	}
 	constructor(props: DummyProps) {
 		super(props);
 		this.state = {
 			input: `x${this.props}x`
 		}
 	}
 
-	render({ initialInput }: DummyProps, { input }: DummyState) {
-		return <DummerComponent initialInput={initialInput} input={input} />
+	render(props: DummyProps, state: DummyState) {
+		const { initialInput } = this.props;
+		const { input } = this.state;
+		return (
+			<DummerComponent
```

Comment:
Now I'm _really_ nitpicking, but it'd be a little less jarring if the typo ("Dummer") were corrected here, too (to "Dumber"). Not a big deal, though.